website/content/docs/operations/stateful-workloads.mdx

```diff
@@ -0,0 +1,116 @@
+
+This availability is useful when sharing storage across multiple instances, such as shared files, or quickly accessing data during failover. The downside is that it can lead to contention issues if not properly managed by the consuming application. For example, storing PostgreSQL's database files on an NFS share and running three instances of PostgreSQL simultaneously accessing the same NFS can lead to data corruption. 
+
+### Private cloud or on-premises
+
+When running workloads on-premises in a self-managed private cloud, SAN/NAS systems or Software Defined Storage like Portworx or Ceph usually provisions non-local storage. The storage can be accessed using a block protocol such as iSCSI, FC, NVMe-oF, or a file protocol such as NFS, CIFS, or sometimes both. Dedicated storage teams manage these systems in most organizations.
+
+## Persistent storage options
+
+The following are options for persistent storage. Since environments differ depending on application requirements, it is important to choose the appropriate storage for your environment. The most important things to consider are performance, reliability, availability, and maintenance.
+
+### CSI
+
+[Container Storage Interface](https://github.com/container-storage-interface/spec) is a vendor-neutral specification that allows storage providers to develop plugins that orchestrators such as Nomad can use. CSI plugins can support various features, such as dynamic provisioning of volumes and their lifecycle management, such as snapshots, deletion, and dynamic resizing. The exact feature set will depend on the plugin and the underlying storage platform.  
+
+You can find a list of plugins and their feature set in the [Kubernetes CSI Developer Documentation](https://kubernetes-csi.github.io/docs/drivers.html)
+
+While Nomad follows the CSI specification, some plugins may implement orchestrator-specific logic that makes them incompatible with Nomad. You should validate that your chosen plugin works with Nomad before using it. Refer to the plugin documentation for more information.
+
+There are three CSI plugin subtypes:
+
+- Controller, which communicates with the storage provider for the management of the volume lifecycle
+- Node, which runs on all Nomad clients and handles all local operations (for example, mounting/unmounting volumes in allocations) and have to be `privileged` to be able to perform those operations
+- Monolithic, which combines both the above roles.
+
+All types can and should be run as Nomad jobs - `system` jobs for Node and Monolithic, `service` for Controllers.
```

Comment:
This is heavily covered in https://nomad-kfsnumtbi-hashicorp.vercel.app/nomad/docs/concepts/plugins/csi so we should at the very least link to it from the relevant sections.

Reply:
Yup I'll add it.